sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2342,6 +2342,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_CENTRAL_MOMENT_AGG =
+    buildConf("spark.sql.legacy.centralMomentAgg")
+      .internal()
+      .doc("When set to true, central moment aggregation will return Double.NaN " +
+        "if divide by zero occurred during calculation. Otherwise, it will return null. " +
+        "Before version 3.1.0, it returns NaN in divideByZero case by default.")
```

Comment:
Shall we update migration guide (https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md) too? Looks like it's more a behaviour change instead of a big fix.

Reply:
Ok, will update.